Library/Homebrew/test/formula_creator_spec.rb

```diff
@@ -3,28 +3,61 @@
 require "formula_creator"
 
 RSpec.describe Homebrew::FormulaCreator do
-  it "gets name from GitHub archive URL" do
-    t = described_class.name_from_url("https://github.com/abitrolly/lapce/archive/v0.3.0.tar.gz")
-    expect(t).to eq("lapce")
-  end
-
-  it "gets name from gitweb URL" do
-    t = described_class.name_from_url("http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary")
-    expect(t).to eq("libzipper")
-  end
-
-  it "gets name from GitHub repo URL" do
-    t = described_class.name_from_url("https://github.com/abitrolly/lapce.git")
-    expect(t).to eq("lapce")
-  end
-
-  it "gets name from GitHub download URL" do
-    t = described_class.name_from_url("https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz")
-    expect(t).to eq("stella")
-  end
+  describe ".new" do
+    tests = {
+      "generic tarball URL": {
+        params:   {
+          url: "http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz",
+        },
+        expected: {
+          name:    "synscan",
+          version: "5.02",
+        },
+      },
+      "gitweb URL":          {
+        params:   {
+          url: "http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary",
+        },
+        expected: {
+          name: "libzipper",
+        },
+      },
+      "GitHub repo URL":     {
+        params:   {
+          url: "https://github.com/abitrolly/lapce.git",
+        },
+        expected: {
+          name: "lapce",
+        },
+      },
+      "GitHub archive URL":  {
+        params:   {
+          url: "https://github.com/abitrolly/lapce/archive/v0.3.0.tar.gz",
+        },
+        expected: {
+          name:    "lapce",
+          version: "0.3.0",
+        },
+      },
+      "GitHub download URL": {
+        params:   {
+          url: "https://github.com/stella-emu/stella/releases/download/6.7/stella-6.7-src.tar.xz",
+        },
+        expected: {
+          name:    "stella",
+          version: "6.7",
+        },
+      },
+    }
 
-  it "gets name from generic tarball URL" do
-    t = described_class.name_from_url("http://digit-labs.org/files/tools/synscan/releases/synscan-5.02.tar.gz")
-    expect(t).to eq("synscan")
+    tests.each do |description, test|
+      it "parses #{description}" do
+        params = test[:params]
+        fc = described_class.new(**params)
```

Comment:
```suggestion
        fc = described_class.new(url: test[:params][:url])
```
as you never pass any other parameters. would also probably be nice to flatten out `params` given that.